File: ui/trading-bot-ui/src/components/TradeJournal.js
import { useEffect, useState } from "react";
const BASE_URL = process.env.REACT_APP_API_URL;
function TradeJournal({ userId }) {
    const [journal, setJournal] = useState([]);

    useEffect(() => {
        fetch(`${BASE_URL}/api/trade_journal?user_id=${userId}`)
            .then(response => response.json())
            .then(data => setJournal(data))
            .catch(error => console.error("Error fetching trade journal:", error));
    }, [userId]);

    return (
        <div>
            <h2>AI Trade Journal</h2>
            <table>
                <thead>
                    <tr>
                        <th>Symbol</th>
                        <th>Trade Type</th>
                        <th>AI Confidence (%)</th>
                        <th>Execution Status</th>
                    </tr>
                </thead>
                <tbody>
                    {journal.map((entry, index) => (
                        <tr key={index}>
                            <td>{entry.symbol}</td>
                            <td>{entry.trade_type}</td>
                            <td>{entry.ai_confidence}%</td>
                            <td>{entry.execution_status}</td>
                        </tr>
                    ))}
                </tbody>
            </table>
        </div>
    );
}

export default TradeJournal;

File: services/trade_journal.py
from sqlalchemy.orm import Session
from database.models import AITradeJournal

def save_trade_to_journal(user_id: int, symbol: str, trade_type: str, ai_confidence: float, status: str, db: Session):
    """Save AI trade decisions to the journal for analysis."""
    
    trade_entry = AITradeJournal(
        user_id=user_id,
        symbol=symbol,
        trade_type=trade_type,
        ai_confidence=ai_confidence,
        execution_status=status
    )
    
    db.add(trade_entry)
    db.commit()
